fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuExperimentsRobot.kt

```diff
@@ -0,0 +1,55 @@
+package org.mozilla.fenix.ui.robots
+
+import android.view.View
+import android.widget.TextView
+import androidx.compose.ui.platform.ComposeView
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.uiautomator.UiSelector
+import org.junit.Assert.assertTrue
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.TestAssetHelper.waitingTime
+import org.mozilla.fenix.helpers.TestHelper.mDevice
+import org.mozilla.fenix.helpers.click
+
+/**
+ * Implementation of Robot Pattern for the experiments sub menu.
+ */
+class SettingsSubMenuExperimentsRobot {
+
+    class Transition {
+
+        fun goBackToHomeScreen(interact: HomeScreenRobot.() -> Unit): HomeScreenRobot.Transition {
+            goBackButton().click()
+
+            HomeScreenRobot().interact()
+            return HomeScreenRobot.Transition()
+        }
+
+        fun goBack(interact: SettingsRobot.() -> Unit): SettingsRobot.Transition {
+            goBackButton().click()
+
+            SettingsRobot().interact()
+            return SettingsRobot.Transition()
+        }
+
+    }
+
+    private fun getExperiment(experimentName: String): UiSelector? {
+        return UiSelector().textContains(experimentName)
+    }
+
+    fun verifyExperimentExists(title: String) {
+        val experiment = getExperiment(title)
+
+        if (experiment != null) {
```

Comment:
In the `getExperiment` function, you can simplify this in Kotlin with

```Kotlin
return try {
    checkNotNull(UiSelector().textContains(experimentName)) { "UiSelector().textContains(experimentName) must not be null" }
} catch (exception: IllegalStateException) {
    // handle the exception
    "Default value"
}

```

Which eliminates the need for the `!=` null check here.

Reply:
This is nice, do you think I need to handle the exception? I think it should just error and fail the test

Reply:
I'm fine either way. I don't know the feature at all really, i.e, if there's ever a case where the experiment name doesn't show up or not and how Fenix would showcase it.